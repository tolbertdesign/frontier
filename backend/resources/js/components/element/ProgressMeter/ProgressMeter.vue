<template>
  <div :class="['meter', backgroundColorClass, (showValue && firstMeterPercentage === 0 ? 'noValueDisplay pb-6': '')]">
    <span
      :style="{ width: firstMeterPercentage + '%' }"
      :class="['meter-1', color === 'secondary' ? 'has-background-secondary': 'bg-primary' ]"
    />
    <span
      v-if="showValue"
      :style="getPercentagePosition(firstMeterPercentage)"
      class="valueDisplay font-semibold"
    >
      {{ valueToDisplay }}
    </span>
    <span
      :style="{ width: secondMeterPercentage + '%' }"
      class="meter-2"
    />
  </div>
</template>

<script>
export default {
  name: 'ProgressMeter',
  status: 'prototype',
  release: '1.0.0',
  props: {
    color: {
      type: String,
      default: 'primary',
    },
    backgroundColorClass: {
      type: String,
      default: 'bg-grey-light',
    },
    goal: {
      type: [String, Number],
      default: 100,
    },
    firstMeter: {
      type: Number,
      default: 0,
    },
    secondMeter: {
      type: Number,
      default: 0,
    },
    showValue: {
      type: Boolean,
      default: false,
    },
    valueToDisplay: {
      type: String,
      default: '',
    },
  },
  computed: {
    firstMeterPercentage () {
      return this.getPercentage(this.firstMeter, this.goal)
    },
    secondMeterPercentage () {
      return this.getPercentage(this.firstMeter + this.secondMeter, this.goal)
    },
  },
  methods: {
    getPercentage (part, total) {
      if (isNaN(part) || isNaN(total) || total <= 0) {
        return 0
      }
      const percent = part / total * 100
      if (percent > 100) {
        return 100
      }
      return percent
    },
    getPercentagePosition (percent) {
      if (percent > 15) {
        return { right: Math.abs((percent - 100)) + '%' }
      } else {
        return { left: '0%' }
      }
    },
  },
}
</script>
