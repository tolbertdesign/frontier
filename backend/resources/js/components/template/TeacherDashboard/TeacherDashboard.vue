<template>
  <div class="mx-4 lg:mx-0 mb-16 lg:mb-8">
    <div class="flex flex-col rounded-lg lg:flex-row lg:border lg:shadow">
      <div
        v-if="!program.hide_teacher_action_steps"
        class="w-full mb-6 border rounded-lg shadow lg:mb-0 lg:w-1/2 lg:rounded-none lg:shadow-none lg:border-t-0 lg:border-b-0 lg:border-l-0"
      >
        <TeacherActionSteps
          :program="program"
          :tasks="teacherParticipant.user_tasks"
        />
      </div>
      <div
        :class="{ 'lg:w-full': program.hide_teacher_action_steps }"
        class="w-full border rounded-lg shadow lg:w-1/2 lg:border-none lg:shadow-none"
      >
        <TeacherPledgeMeter :program="program" />
      </div>
    </div>
  </div>
</template>

<script>
import TeacherActionSteps from '@/components/template/TeacherActionSteps'
import TeacherPledgeMeter from '@/components/template/TeacherPledgeMeter'

export default {
  name: 'TeacherDashboardComponent',
  status: 'prototype',
  release: '1.0',
  components: {
    TeacherActionSteps,
    TeacherPledgeMeter,
  },
  props: {
    program: {
      type: Object,
      default: null,
    },
  },
  computed: {
    teacherParticipant () {
      return this.$store.getters.teacherParticipant(this.program)
    },
  },
}
</script>
