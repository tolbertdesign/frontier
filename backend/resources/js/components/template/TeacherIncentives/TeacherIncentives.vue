<template>
  <div class="text-center">
    <!-- Teacher incentive image for mobile -->
    <img
      :src="`https://${this.$store.state.s3Bucket}.s3.amazonaws.com/assets/images/teacher_incentives_mobile.png`"
      class="p-2 md:hidden"
    >

    <!-- Teacher incentive image for desktop -->
    <img
      :src="`https://${this.$store.state.s3Bucket}.s3.amazonaws.com/assets/images/teacher_incentives_desktop.png`"
      class="p-2 md:inline hidden"
    >
  </div>
</template>

<script>
export default {
  name: 'TeacherIncentives',
  status: 'prototype',
  release: '1.0.0',
}
</script>
