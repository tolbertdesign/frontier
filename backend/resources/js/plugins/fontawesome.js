import Vue from 'vue'

import { library, dom } from '@fortawesome/fontawesome-svg-core'

// Brands (fab)
import { faFacebook, faFacebookF, faFacebookSquare, faTwitter } from '@fortawesome/free-brands-svg-icons'

// Regular (far)
import {
  faAddressCard,
  faArrowCircleRight,
  faBars as farBars,
  faCircle,
  faCommentAltLines,
  faEdit,
  faEnvelope,
  faExternalLinkSquare,
  faInfoCircle,
  faLock,
  faSignIn,
  faSignOut,
  faSquare,
  faTimes as farTimes,
  faUserEdit as farUserEdit,
} from '@fortawesome/pro-regular-svg-icons'

// Light (fal)
import { faBars, faTimes, faUserEdit } from '@fortawesome/pro-light-svg-icons'

// Solid Style Icons (fas)
import {
  faBars as fasBars,
  faBell,
  faBold,
  faCaretDown,
  faCaretUp,
  faCheck,
  faCheckSquare,
  faChevronDown,
  faChevronUp,
  faChevronLeft,
  faChevronRight,
  faCircle as fasCircle,
  faCode,
  faCoffee,
  faCog,
  faComment,
  faEdit as fasEdit,
  faEnvelope as fasEnvelope,
  faExclamationCircle,
  faExclamationTriangle,
  faExpandArrowsAlt,
  faEye,
  faItalic,
  faLink,
  faListOl,
  faListUl,
  faLock as fasLock,
  faMinus,
  faMobileAlt,
  faParagraph,
  faPlayCircle,
  faPlus,
  faQuestionCircle,
  faQuoteRight,
  faRedo,
  faShare,
  faShareSquare,
  faSortDown,
  faSpinner,
  faStrikethrough,
  faTimes as fasTimes,
  faTrashAlt,
  faUnderline,
  faUndo,
  faUser,
  faUsers,
  faUserPlus,
} from '@fortawesome/free-solid-svg-icons'

import {
  FontAwesomeIcon,
  FontAwesomeLayers,
  FontAwesomeLayersText,
} from '@fortawesome/vue-fontawesome'

library.add(
  faAddressCard,
  faArrowCircleRight,
  faBars,
  faBell,
  faBold,
  faCaretDown,
  faCaretUp,
  faCheck,
  faCheckSquare,
  faChevronDown,
  faChevronUp,
  faChevronLeft,
  faChevronRight,
  faCircle,
  faCode,
  faCoffee,
  faCog,
  faComment,
  faCommentAltLines,
  faEdit,
  fasEdit,
  faEnvelope,
  faExclamationCircle,
  faExclamationTriangle,
  faExpandArrowsAlt,
  faExternalLinkSquare,
  faEye,
  faFacebook,
  faFacebookF,
  faFacebookSquare,
  faInfoCircle,
  faItalic,
  faLink,
  faListOl,
  faListUl,
  faLock,
  faMinus,
  faMobileAlt,
  faParagraph,
  faPlayCircle,
  faPlus,
  faQuestionCircle,
  faQuoteRight,
  faRedo,
  farBars,
  farTimes,
  fasBars,
  fasCircle,
  fasEnvelope,
  faShare,
  faShareSquare,
  faSignIn,
  faSignOut,
  fasLock,
  faSortDown,
  faSpinner,
  faSquare,
  fasTimes,
  faUsers,
  faStrikethrough,
  faTimes,
  faTrashAlt,
  faTwitter,
  faUnderline,
  faUndo,
  faUser,
  faUserEdit,
  faUserPlus,
  farUserEdit,
)

dom.watch()

Vue.component('FontAwesomeIcon', FontAwesomeIcon)
Vue.component('FontAwesomeLayers', FontAwesomeLayers)
Vue.component('FontAwesomeLayersText', FontAwesomeLayersText)
