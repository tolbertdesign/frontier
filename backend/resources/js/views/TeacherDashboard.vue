<template>
  <ul class="page min-h-screen">
    <li
      v-for="(program, index) in teacherPrograms"
      :key="index"
    >
      <section
        :class="program.microsite.microsite_color_theme.theme_name"
        class="page min-h-screen"
      >
        <AppHeader
          :program="program"
          :index="index"
        >
          <ProgramHeader :program="program" />
        </AppHeader>

        <TeacherDashboardLayout
          :program="program"
          class="pt-24 lg:pt-52"
        />
      </section>
    </li>
  </ul>
</template>

<script>
import TeacherDashboardLayout from '@/layouts/TeacherDashboardLayout'
import AppHeader from '@/components/template/AppHeader'
import ProgramHeader from '@/components/pattern/ProgramHeader'
import { mapGetters } from 'vuex'

export default {
  name: 'TeacherDashboardView',
  status: 'prototype',
  release: '1.0.0',
  components: {
    TeacherDashboardLayout,
    AppHeader,
    ProgramHeader,
  },
  computed: {
    ...mapGetters(['activePrograms']),
    metaInfo () {
      return {
        title: this.$store.state.activeEventName || this.activePrograms[0].event_name,
      }
    },
    teacherPrograms () {
      const programs = []
      this.activePrograms.forEach(program => {
        program.participants.forEach(participant => {
          if (participant.id === this.$store.state.User.teacher_participant_id) {
            programs.push(program)
          }
        })
      })

      return programs
    },
  },
}
</script>
