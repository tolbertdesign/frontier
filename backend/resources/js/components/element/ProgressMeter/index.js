import ProgressMeter from './ProgressMeter.vue'

export default ProgressMeter
