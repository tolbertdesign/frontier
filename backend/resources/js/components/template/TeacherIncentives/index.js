import TeacherIncentives from './TeacherIncentives'

export default TeacherIncentives
