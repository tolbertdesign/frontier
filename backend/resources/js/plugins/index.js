import 'nprogress/nprogress.css'
// import '@fortawesome/fontawesome-free/css/all.min.css'

import '@/plugins/buefy'
import '@/plugins/fontawesome'
import '@/plugins/gravatar'
import '@/plugins/tooltips'
import '@/plugins/vue-croppa'
import '@/plugins/vue-meta'
import '@/plugins/vuelidate'

// import '@/plugins/validation'
