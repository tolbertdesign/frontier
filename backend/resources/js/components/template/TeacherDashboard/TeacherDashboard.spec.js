import { shallowMount, createLocalVue } from '@vue/test-utils'
import TeacherDashboard from './TeacherDashboard.vue'
import TeacherIncentives from '@/components/template/TeacherIncentives'
import ProgressMeter from '@/components/element/ProgressMeter'
import Buefy from 'buefy'
import { FontAwesomeIcon } from '@fortawesome/vue-fontawesome'
import '@/plugins/fontawesome'

const localVue = createLocalVue()
localVue.use(Buefy, {
  defaultIconPack: 'fas',
})

const program = {
  participants: [
    {
      profile: {
        image_name: 'imageName',
      },
    },
  ],
}

const $store = {
  getters: {
    teacherParticipant: (program) => {
      return program.participants[0]
    },
  },
}

describe('TeacherDashboard', () => {
  it('Snapshop test for TeacherDashboard', () => {
    const wrapper = shallowMount(TeacherDashboard, {
      localVue,
      mocks: { $store },
      components: {
        TeacherIncentives,
        ProgressMeter,
        FontAwesomeIcon,
      },
      propsData: {
        program: program,
      },
    })
    expect(wrapper).toMatchSnapshot()
  })
})
