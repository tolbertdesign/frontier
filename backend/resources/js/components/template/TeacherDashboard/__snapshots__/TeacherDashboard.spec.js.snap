// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeacherDashboard Snapshop test for TeacherDashboard 1`] = `
<div class="mx-4 lg:mx-0 mb-16 lg:mb-8">
  <div class="flex flex-col rounded-lg lg:flex-row lg:border lg:shadow">
    <div class="w-full mb-6 border rounded-lg shadow lg:mb-0 lg:w-1/2 lg:rounded-none lg:shadow-none lg:border-t-0 lg:border-b-0 lg:border-l-0">
      <teacheractionsteps-stub program="[object Object]"></teacheractionsteps-stub>
    </div>
    <div class="w-full border rounded-lg shadow lg:w-1/2 lg:border-none lg:shadow-none">
      <teacherpledgemeter-stub program="[object Object]"></teacherpledgemeter-stub>
    </div>
  </div>
</div>
`;
